src/AltinnCore/Designer/Controllers/RuntimeAPIController.cs

```diff
@@ -232,25 +233,88 @@ public FileResult GetInstanceFromFile(string org, string service, string develop
-            string fileToWriteTo = string.Format(pathToSaveTo, partyId, instanceId, attachmentType, guid.ToString() + fileExtension);
+            string pathToSaveTo = _settings.GetTestdataForPartyPath(org, service, developer) + "{0}/{1}/data/{2}/{3}/{4}";
+            Directory.CreateDirectory(string.Format(pathToSaveTo, partyId, instanceId, attachmentType, guid, string.Empty));
+            string fileToWriteTo = string.Format(pathToSaveTo, partyId, instanceId, attachmentType, guid, attachmentName);
             using (Stream streamToWriteTo = System.IO.File.Open(fileToWriteTo, FileMode.OpenOrCreate))
             {
                 await Request.StreamFile(streamToWriteTo);
                 streamToWriteTo.Flush();
             }
 
+            return Ok(new { id = guid });
+        }
+
+        /// <summary>
+        /// Method that removes a form attachment from designer disk
+        /// </summary>
+        /// <param name="org">The organization for the service</param>
+        /// <param name="service">The name of the service</param>
+        /// <param name="developer">The current developer</param>
+        /// <param name="partyId">The party id of the test user</param>
+        /// <param name="instanceId">The instance id</param>
+        /// <param name="attachmentType">The attachment type id</param>
+        /// <param name="attachmentId">The attachment id</param>
+        /// <returns>The status of the deletion</returns>
+        [HttpPost]
+        [DisableFormValueModelBinding]
+        public IActionResult DeleteFormAttachment(string org, string service, string developer, int partyId, Guid instanceId, string attachmentType, string attachmentId)
+        {
+            string pathToDelete = _settings.GetTestdataForPartyPath(org, service, developer) + "{0}/{1}/data/{2}/{3}/";
+            DirectoryInfo directory = new DirectoryInfo(string.Format(pathToDelete, partyId, instanceId, attachmentType, attachmentId));
+            foreach (FileInfo file in directory.EnumerateFiles())
+            {
+                file.Delete();
+            }
+
+            directory.Delete();
             return Ok();
         }
 
+        /// <summary>
+        /// Method that gets metadata on form attachments form designer disk
+        /// </summary>
+        /// <param name="org">The organization for the service</param>
+        /// <param name="service">The name of the service</param>
+        /// <param name="developer">The current developer</param>
+        /// <param name="partyId">The party id of the test user</param>
+        /// <param name="instanceId">The instance id</param>
+        /// <returns>A list with attachments metadata ordered by attachmentType</returns>
+        [HttpGet]
+        [DisableFormValueModelBinding]
+        public IActionResult GetFormAttachments(string org, string service, string developer, int partyId, Guid instanceId)
```

Comment:
Maybe we should add some model for 
ArrayList allAttachments = new ArrayList(); and
ArrayList attachments = new ArrayList();
so they can be easily reused later

Reply:
Agreed, added Attachment and AttachmentList to our common models.